src/librustc/traits/select.rs

```diff
@@ -3340,11 +3339,26 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
             source, target
         );
 
+        let cause = ObligationCause::new(
+            obligation.cause.span,
+            obligation.cause.body_id,
+            ObjectCastObligation(target),
+        );
+
+        let predicate_to_obligation = |predicate| {
```

Comment:
Could use a comment I think.

Reply:
Was pre-existing, and is pretty self-explanatory what it does anyway so...

Reply:
If I ask for a comment it means it is not self explanatory for me =P